include/deal.II/grid/grid_tools_cache.h

```diff
@@ -76,7 +90,16 @@ namespace GridTools
      * @param mapping The mapping to use when computing cached objects
      */
     Cache (const Triangulation<dim,spacedim> &tria,
-           const Mapping<dim,spacedim> &mapping=StaticMappingQ1<dim,spacedim>::mapping);
+           const Mapping<dim,spacedim> &mapping=StaticMappingQ1<dim,spacedim>::mapping
+#ifdef DEAL_II_WITH_MPI
+           ,
+           MPI_Comm mpi_comm=MPI_COMM_WORLD,
```

Comment:
This is bad style -- you make programs that use and don't use MPI incompatible. The better way is to ensure that the signature of functions the same whether or not you use MPI, and simply ignore arguments that are not relevant in some context. (Possibly with an assertion that they are at a value that makes sense in the non-MPI case.)

Reply:
Second, doesn't the communicator have to match the communicator used in the triangulation? If so, get it from there and avoid the potential mistake of passing a mismatched communicator.